### PR TITLE
add templates for azure machine learning service

### PR DESCRIPTION
**Pull request by keerthiadu**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 101-machine-learning-service-create/prereqs/prereq.azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            "sku":{
+               "name":"standard",
+               "family":"A"
+            },
+            "accessPolicies":[]
+         }
+      },
+      {
+         "type":"Microsoft.Insights/components",
+         "apiVersion":"2018-05-01-preview",
```
Comment (bmoore-msft) [inline 101-machine-learning-service-create/prereqs/prereq.azuredeploy.json:88]: update apiVersion

File: 101-machine-learning-service-create/metadata.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "validationType": "Manual",
```
Comment (bmoore-msft) [inline 101-machine-learning-service-create/metadata.json:4]: why does this require manual validation?

File: 101-machine-learning-service-create/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,288 @@
+{
+   "$schema":"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+   "contentVersion":"1.0.0.0",
+   "parameters":{
+      "webserviceName":{
```
Comment (bmoore-msft) [inline 101-machine-learning-service-create/azuredeploy.json:5]: add metadata descriptions to all parameters

File: 101-machine-learning-service-create/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+            "name":"models",
+            "count":"[length(parameters('models'))]",
+            "input":{
+               "name":"[parameters('models')[copyIndex('models')].name]",
+               "url":"[if(startsWith(parameters('models')[copyIndex('models')].path, '/'), concat('aml://storage', parameters('models')[copyIndex('models')].path), concat('aml://storage/', parameters('models')[copyIndex('models')].path))]",
+               "mimeType":"[parameters('models')[copyIndex('models')].mimeType]",
+               "framework":"[parameters('models')[copyIndex('models')].framework]",
+               "frameworkVersion":"[parameters('models')[copyIndex('models')].frameworkVersion]"
+            },
+            "mode":"serial"
```
Comment (bmoore-msft) [inline 101-machine-learning-service-create/azuredeploy.json:250]: move this above input

File: 101-machine-learning-service-create/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+         "defaultValue":[
+            
+         ],
+         "metadata":{
+            "description":"The IDs of the model to be deployed. Model Id takes the form of <model name>:<model version>. Models need to be registered first."
+         }
+      },
+      "models":{
+         "type":"array",
+         "defaultValue":[
```
Comment (bmoore-msft) [inline 101-machine-learning-service-create/azuredeploy.json:103]: can we add a non-null defaultValue so we can test the looping code?

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 101-machine-learning-service-create/prereqs/prereq.azuredeploy.json (hunk 1 above)
Comment (keerthiadu) [reply 101-machine-learning-service-create/prereqs/prereq.azuredeploy.json:88]: made changes

File: 101-machine-learning-service-create/azuredeploy.json (hunk 3 above)
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json:5]: addeed

File: 101-machine-learning-service-create/azuredeploy.json (hunk 4 above)
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: made changes

File: 101-machine-learning-service-create/metadata.json (hunk 2 above)
Comment (keerthiadu) [reply 101-machine-learning-service-create/metadata.json:4]: inputs for modelIds and/or models are very very specific to the user's machine learning models and we cant provide a default value.

File: 101-machine-learning-service-create/azuredeploy.json (hunk 5 above)
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json:109]: inputs for modelIds and/or models are very very specific to the user's machine learning models and unfortunately we cant provide a non-null default value.

File: 101-machine-learning-service-create/metadata.json (hunk 2 above)
Comment (bmoore-msft) [reply 101-machine-learning-service-create/metadata.json:4]: We'll need to be able to validate it one way or another - so if we can't automate it we'll need instructions on how to do it manually.

File: 101-machine-learning-service-create/azuredeploy.json (hunk 5 above)
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: removed the default value

File: 101-machine-learning-service-create/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,294 @@
+{
+   "$schema":"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+   "contentVersion":"1.0.0.0",
+   "parameters":{
+      "webserviceName":{
+         "type":"string",
+         "metadata":{
+            "description":"The name of the Azure Machine Learning Web Service."
```
[the remark this answers is not in the document]
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: made changes

File: 101-machine-learning-service-create/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+         "defaultValue":"[resourceGroup().location]",
+         "metadata":{
+            "description":"The location of the Azure Machine Learning Workspace."
+         }
+      },
+      "environmentName":{
+         "type":"string",
+         "defaultValue":"",
+         "metadata":{
+            "description":"Name of Azure Machine Learning Environment for deployment. See https://docs.microsoft.com/en-us/azure/machine-learning/how-to-use-environments and https://docs.microsoft.com/en-us/azure/machine-learning/resource-curated-environments ."
```
[the remark this answers is not in the document]
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: removed the default value

File: 101-machine-learning-service-create/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+         }
+      },
+      "environmentVersion":{
+         "type":"string",
+         "defaultValue":"",
+         "metadata":{
+            "description":"Version of Azure Machine Learning Environment for deployment."
```
[the remark this answers is not in the document]
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: removed the default value

File: 101-machine-learning-service-create/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+         "type":"string",
+         "metadata":{
+            "description":"The amount of memory (in GB) to allocate for this Webservice. Can be a decimal."
+         }
+      },
+      "driverProgram":{
+         "type":"string",
+         "defaultValue":"",
+         "metadata":{
+            "description":"Relative path of a file from storage account that contains the code to run for service."
```
[the remark this answers is not in the document]
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: removed the default value

File: 101-machine-learning-service-create/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+            "description":"The number of containers to allocate for this Webservice. No default, if this parameter is not set then the autoscaler is enabled by default. Only applicable when deploying to AKS."
+         }
+      },
+      "modelIds":{
+         "type":"array",
+         "defaultValue":[
+            
+         ],
+         "metadata":{
+            "description":"The IDs of the model to be deployed. Model Id takes the form of <model name>:<model version>. Models need to be registered first."
```
[the remark this answers is not in the document]
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: removed the default value

File: 101-machine-learning-service-create/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+         "defaultValue":false,
+         "metadata":{
+            "description":"Whether or not to enable token auth for this Webservice.  Only applicable when deploying to AKS."
+         }
+      },
+      "primaryKey":{
+         "type":"securestring",
+         "defaultValue":"",
+         "metadata":{
+            "description":"A primary auth key to use for this Webservice."
```
[the remark this answers is not in the document]
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: removed the default value

File: 101-machine-learning-service-create/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+         }
+      },
+      "secondaryKey":{
+         "type":"securestring",
+         "defaultValue":"",
+         "metadata":{
+            "description":"A secondary auth key to use for this Webservice."
```
[the remark this answers is not in the document]
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: removed the default value

File: 101-machine-learning-service-create/azuredeploy.json [hunk 13]
```diff
@@ -0,0 +1,294 @@
...
+      "webserviceName":{
+         "type":"string",
+         "metadata":{
+            "description":"The name of the Azure Machine Learning Web Service."
+         }
+      },
+      "workspaceName":{
+         "type":"string",
+         "metadata":{
+            "description":"The name of the Azure Machine Learning Workspace."
```
[the remark this answers is not in the document]
Comment (keerthiadu) [reply 101-machine-learning-service-create/azuredeploy.json]: added

**Comment by keerthiadu** [comment]
CI gates failed as I can't use the same PR for adding templates for more than one scenario. Created a separate pull request [8518  ](https://github.com/Azure/azure-quickstart-templates/pull/8518)for deploying machine learning service using AKS.

**Pull request merged**